docs/api-reference/test-utils/browser-driver.md

```diff
@@ -6,115 +6,108 @@
-Returns: a `Promise` that resolves when the browser code calls the function to the string the browser script passed to the exposed function.
+* `options` (Object)
+  - `url` (String) - The url to load in the page. Default `http://localhost`.
+  - `exposeFunctions` (Object) - keys are function names to be added to the page's `window` object, and the values are callback functions to execute in Node.js. See [exposeFunction](https://github.com/GoogleChrome/puppeteer/blob/v1.11.0/docs/api.md#pageexposefunctionname-puppeteerfunction) for details.
+  - `onLoad` (Function) - callback when the page is loaded
+  - `onConsole` (Function) - callback when the page logs to console
+  - `onError` (Function) - callback when the page crashes
 
+Returns a `Promise` that resolves when the page is open.
 
-### stopBrowser
 
-Stops and closes the browser.
+### stopBrowser()
 
-`driver.stopBrowser()`
+Terminate the browser instance.
 
 Returns a `Promise` that resolves when the browser is closed.
 
 
-### setShellStatus
-
-`driver.setShellStatus(success)`
-
-Set the return value that will be visible to the shell, truthy values will generate 0 which represents success.
-
-
-### startServer
+### startServer(config)
 
 Runs a server in a new child process, that the browser can connect to.
 
-`driver.startServer({process, parameters, options})`
-
-* `process`=`'./node_modules/.bin/webpack-dev-server'` - The process to fork
-* `parameters`=`['--config', 'test/render/webpack.config.js']` - The parameters to the process
-* `options`=`{maxBuffer: 5000 * 1024}`
-
-Example:
 ```js
 driver.startServer({
-  process: './node_modules/.bin/webpack-dev-server',
-  parameters: ['--config', 'test/render/webpack.config.js'],
-  options: {maxBuffer: 5000 * 1024}
+  command: './node_modules/.bin/webpack-dev-server',
+  arguments: ['--config', 'test/webpack.config.js'],
+  wait: 2000
 })
 ```
 
-Returns: nothing
+Parameters:
 
+* `command` (String) - the command to run, default `'webpack-dev-server'`.
+* `arguments` (Array<String>) - a list of string arguments.
+* `options` (Object) - options for the new process. Default `{maxBuffer: 5000 * 1024}`. See [child_process.spawn](https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options) for details.
+* `port` (`'auto'`|`false`) - `startServer` can attempt to bind the service to an available port if `port` is set to `'auto'`. In this case, the command receives additional arguments `--port <port>`. Default `'auto'`.
+* `wait` (Number) - time in milliseconds to wait after executing the command. If any error is generated from the child process during this period, the `Promise` will reject. Otherwise, the service is considered available. Default `2000`.
 
-### stopServer
+Returns a `Promise` that resolves to the service URL when the server is available.
 
-Stopsthe server (kills the child process)
 
-`driver.stopServer()`
+### stopServer()
 
-Returns: nothing
+Stops the server (kills the child process).
```

Comment:
Nit: Does it actually kill the child process? It should but I think have had problems with this.

Reply:
It does if you reach this method, or manually kill the browser test process (Ctrl+C). If the browser test script crashes then no. We need to listen to the process events and handle that.